From d1ed6d27dd73a300b94237f4664fb27bd978dbbd Mon Sep 17 00:00:00 2001
Message-Id: <d1ed6d27dd73a300b94237f4664fb27bd978dbbd@dist-git>
From: John Ferlan <jferlan@redhat.com>
Date: Mon, 25 Jul 2016 12:42:51 -0400
Subject: [PATCH] tests: Adjust LUKS tests to use 'volume' secret type

https://bugzilla.redhat.com/show_bug.cgi?id=1301021

Commit id's '9bbf0d7e6' and '2552fec24' added some XML parsing tests
for a LUKS volume to use a 'passphrase' secret format. After commit,
this was deemed to be incorrect, so covert the various tests to use
the volume usage format where the 'usage' is the path to the volume
rather than a user defined name string.

Also, removed the qemuxml2argv-luks-disk-cipher.xml since it was
just a duplicate of qemuxml2argv-luks-disks.xml.

Signed-off-by: John Ferlan <jferlan@redhat.com>
(cherry picked from commit a8d0afc75a0a5c902f1e4188a616651a17a9d050)
---
 .../qemuxml2argv-luks-disk-cipher.xml              | 45 ----------------------
 tests/qemuxml2argvdata/qemuxml2argv-luks-disks.xml |  2 +-
 .../qemuxml2xmlout-luks-disk-cipher.xml            |  1 -
 tests/qemuxml2xmltest.c                            |  1 -
 tests/storagevolxml2xmlin/vol-luks-cipher.xml      |  2 +-
 tests/storagevolxml2xmlin/vol-luks.xml             |  2 +-
 tests/storagevolxml2xmlout/vol-luks-cipher.xml     |  2 +-
 tests/storagevolxml2xmlout/vol-luks.xml            |  2 +-
 8 files changed, 5 insertions(+), 52 deletions(-)
 delete mode 100644 tests/qemuxml2argvdata/qemuxml2argv-luks-disk-cipher.xml
 delete mode 120000 tests/qemuxml2xmloutdata/qemuxml2xmlout-luks-disk-cipher.xml

diff --git a/tests/qemuxml2argvdata/qemuxml2argv-luks-disk-cipher.xml b/tests/qemuxml2argvdata/qemuxml2argv-luks-disk-cipher.xml
deleted file mode 100644
index 9ce15c0..0000000
--- a/tests/qemuxml2argvdata/qemuxml2argv-luks-disk-cipher.xml
+++ /dev/null
@@ -1,45 +0,0 @@
-<domain type='qemu'>
-  <name>encryptdisk</name>
-  <uuid>496898a6-e6ff-f7c8-5dc2-3cf410945ee9</uuid>
-  <memory unit='KiB'>1048576</memory>
-  <currentMemory unit='KiB'>524288</currentMemory>
-  <vcpu placement='static'>1</vcpu>
-  <os>
-    <type arch='x86_64' machine='pc-i440fx-2.1'>hvm</type>
-    <boot dev='hd'/>
-  </os>
-  <clock offset='utc'/>
-  <on_poweroff>destroy</on_poweroff>
-  <on_reboot>restart</on_reboot>
-  <on_crash>destroy</on_crash>
-  <devices>
-    <emulator>/usr/bin/qemu</emulator>
-    <disk type='file' device='disk'>
-      <driver name='qemu' type='luks'/>
-      <source file='/storage/guest_disks/encryptdisk'/>
-      <target dev='vda' bus='virtio'/>
-      <encryption format='luks'>
-        <secret type='passphrase' uuid='0a81f5b2-8403-7b23-c8d6-21ccc2f80d6f'/>
-      </encryption>
-      <address type='pci' domain='0x0000' bus='0x00' slot='0x04' function='0x0'/>
-    </disk>
-    <disk type='file' device='disk'>
-      <driver name='qemu' type='luks'/>
-      <source file='/storage/guest_disks/encryptdisk2'/>
-      <target dev='vdb' bus='virtio'/>
-      <encryption format='luks'>
-        <secret type='passphrase' usage='mycluster_myname'/>
-      </encryption>
-      <address type='pci' domain='0x0000' bus='0x00' slot='0x05' function='0x0'/>
-    </disk>
-    <controller type='usb' index='0'>
-      <address type='pci' domain='0x0000' bus='0x00' slot='0x01' function='0x2'/>
-    </controller>
-    <controller type='pci' index='0' model='pci-root'/>
-    <input type='mouse' bus='ps2'/>
-    <input type='keyboard' bus='ps2'/>
-    <memballoon model='virtio'>
-      <address type='pci' domain='0x0000' bus='0x00' slot='0x03' function='0x0'/>
-    </memballoon>
-  </devices>
-</domain>
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-luks-disks.xml b/tests/qemuxml2argvdata/qemuxml2argv-luks-disks.xml
index 9ce15c0..4c9c4c7 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-luks-disks.xml
+++ b/tests/qemuxml2argvdata/qemuxml2argv-luks-disks.xml
@@ -28,7 +28,7 @@
       <source file='/storage/guest_disks/encryptdisk2'/>
       <target dev='vdb' bus='virtio'/>
       <encryption format='luks'>
-        <secret type='passphrase' usage='mycluster_myname'/>
+        <secret type='passphrase' usage='/storage/guest_disks/encryptdisk2'/>
       </encryption>
       <address type='pci' domain='0x0000' bus='0x00' slot='0x05' function='0x0'/>
     </disk>
diff --git a/tests/qemuxml2xmloutdata/qemuxml2xmlout-luks-disk-cipher.xml b/tests/qemuxml2xmloutdata/qemuxml2xmlout-luks-disk-cipher.xml
deleted file mode 120000
index fa55233..0000000
--- a/tests/qemuxml2xmloutdata/qemuxml2xmlout-luks-disk-cipher.xml
+++ /dev/null
@@ -1 +0,0 @@
-../qemuxml2argvdata/qemuxml2argv-luks-disk-cipher.xml
\ No newline at end of file
diff --git a/tests/qemuxml2xmltest.c b/tests/qemuxml2xmltest.c
index d00d209..a757fdb 100644
--- a/tests/qemuxml2xmltest.c
+++ b/tests/qemuxml2xmltest.c
@@ -503,7 +503,6 @@ mymain(void)
     DO_TEST("encrypted-disk");
     DO_TEST("encrypted-disk-usage");
     DO_TEST("luks-disks");
-    DO_TEST("luks-disk-cipher");
     DO_TEST("memtune");
     DO_TEST("memtune-unlimited");
     DO_TEST("blkiotune");
diff --git a/tests/storagevolxml2xmlin/vol-luks-cipher.xml b/tests/storagevolxml2xmlin/vol-luks-cipher.xml
index 009246f..da28a27 100644
--- a/tests/storagevolxml2xmlin/vol-luks-cipher.xml
+++ b/tests/storagevolxml2xmlin/vol-luks-cipher.xml
@@ -15,7 +15,7 @@
       <label>unconfined_u:object_r:virt_image_t:s0</label>
     </permissions>
     <encryption format='luks'>
-      <secret type='passphrase' usage='mumblyfratz'/>
+      <secret type='passphrase' uuid='f52a81b2-424e-490c-823d-6bd4235bc572'/>
       <cipher name='serpent' size='256' mode='cbc' hash='sha256'/>
       <ivgen name='plain64' hash='sha256'/>
     </encryption>
diff --git a/tests/storagevolxml2xmlin/vol-luks.xml b/tests/storagevolxml2xmlin/vol-luks.xml
index eb4dc41..bf3c519 100644
--- a/tests/storagevolxml2xmlin/vol-luks.xml
+++ b/tests/storagevolxml2xmlin/vol-luks.xml
@@ -15,7 +15,7 @@
       <label>unconfined_u:object_r:virt_image_t:s0</label>
     </permissions>
     <encryption format='luks'>
-      <secret type='passphrase' usage='mumblyfratz'/>
+      <secret type='passphrase' uuid='f52a81b2-424e-490c-823d-6bd4235bc572'/>
     </encryption>
   </target>
 </volume>
diff --git a/tests/storagevolxml2xmlout/vol-luks-cipher.xml b/tests/storagevolxml2xmlout/vol-luks-cipher.xml
index 9014849..1ac7424 100644
--- a/tests/storagevolxml2xmlout/vol-luks-cipher.xml
+++ b/tests/storagevolxml2xmlout/vol-luks-cipher.xml
@@ -15,7 +15,7 @@
       <label>unconfined_u:object_r:virt_image_t:s0</label>
     </permissions>
     <encryption format='luks'>
-      <secret type='passphrase' usage='mumblyfratz'/>
+      <secret type='passphrase' uuid='f52a81b2-424e-490c-823d-6bd4235bc572'/>
       <cipher name='serpent' size='256' mode='cbc' hash='sha256'/>
       <ivgen name='plain64' hash='sha256'/>
     </encryption>
diff --git a/tests/storagevolxml2xmlout/vol-luks.xml b/tests/storagevolxml2xmlout/vol-luks.xml
index 5b764b7..7b82866 100644
--- a/tests/storagevolxml2xmlout/vol-luks.xml
+++ b/tests/storagevolxml2xmlout/vol-luks.xml
@@ -15,7 +15,7 @@
       <label>unconfined_u:object_r:virt_image_t:s0</label>
     </permissions>
     <encryption format='luks'>
-      <secret type='passphrase' usage='mumblyfratz'/>
+      <secret type='passphrase' uuid='f52a81b2-424e-490c-823d-6bd4235bc572'/>
     </encryption>
   </target>
 </volume>
-- 
2.9.2

